components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java

```diff
@@ -147,7 +147,11 @@ public UndertowComponent getComponent() {
     }
 
     public UndertowSecurityProvider getSecurityProvider() {
-        return securityProvider;
+        if (this.securityProvider == null) {
```

Comment:
getter/setters should be plain, not these tricks. Instead when creating the endpoint in the component, then set the security provider on the endpoint correctly

Reply:
ok, I'll do it, thanks for explanation